### ISSUE TITLE
bug: `infra.start_infra` throws inside docker container when using SdkDockerClient without docker socket

### ISSUE THREAD

**Issue by sburba** [localstack/localstack#4692]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

`infra.start_infra(asynchronous=True, apis=['s3'])` crashes when run from inside a docker container, SdkDockerClient is used, and the docker socket is not mounted into the container.

### Expected Behavior

`infra.start_infra(asynchronous=True, apis=['s3'])` should ignore the fact that the docker socket is unavailable, as it does when the `LEGACY_DOCKER_CLIENT` environment variable is set to `true`.

### How are you starting LocalStack?

Custom (please describe below)

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)
From inside of a python:3.9 docker container with localstack pip installed, run the following python code:
   ```python
from localstack.services import infra
localstack.start_infra(asynchronous=True, apis=['s3'])
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

   N/A

### Environment

```markdown
- OS: python:3.9 docker image
- LocalStack: 12.17.3
```

### Anything else?

The issue is this line (localstack/utils/docker.py:1096) (in SdkDockerClient:inspect_container)

```python
return self.client().containers.get(container_name_or_id).attrs
```
Will throw `ConnectionError` if it cannot connect to the docker daemon, but only `APIError` and `NotFound` are caught and converted into `ContainerException`s.

Since `ConnectionError` escapes SdkDockerClient:inspect_container without being wrapped in a `ContainerException`, `get_main_container_id` (localstack/utils/bootstrap.py:254) does not properly ignore it and return `None`, causing `print_runtime_information` (localstack/services/infra.py):422 to throw instead of just not printing out the container id.

**Comment by whummer** [comment]
Thanks for reporting @sburba . Do you think you could create a small pull request with the proposed changes? 

Looks like you already have a quite specific solution in mind, which makes the startup code more resilient to situations where the Docker socket is not available - would be great if you could propose the change in a PR, so we can discuss it. Thanks! 🙏

**Comment by sburba** [comment]
I looked into this, looks like it's fixed by upgrading to 0.12.18 of localstack from 0.12.16, as it requires the docker 5.0.0 package which no longer throws ConnectionError in this instance, and instead throws DockerException as localstack expects.
